test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.validation.operations;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.cql3.functions.NativeFunctions;
+import org.apache.cassandra.cql3.functions.NativeScalarFunction;
+import org.apache.cassandra.db.marshal.Int32Type;
+import org.apache.cassandra.db.marshal.VectorType;
+import org.apache.cassandra.transport.ProtocolVersion;
+
+public class CQLVectorTest extends CQLTester.InMemory
+{
+    @Test
+    public void select()
+    {
+        createTable(KEYSPACE, "CREATE TABLE %s (pk vector<int, 2> primary key)");
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 2])");
+
+        Vector<Integer> vector = vector(1, 2);
+        Object[] row = row(vector);
+
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 2]"), row);
```

Comment:
We might want to add the trivial case where there isn't a `WHERE` restriction. While doing that, I have realised that `UntypedResultSet` is missing a `getVector` method. It can be like this:
```java
public <T> List<T> getVector(String column, AbstractType<T> type, int dimension)
{
    ByteBuffer raw = data.get(column);
    return raw == null ? null : VectorType.getInstance(type, dimension).compose(raw);
}
```
And the test in `CQLVectorTest#select` can be:
```java
assertRows(execute("SELECT * FROM %s"), row);
Assertions.assertThat(execute("SELECT * FROM %s").one().getVector("pk", Int32Type.instance, 2))
          .isEqualTo(vector);
```

Reply:
good catch, added